include/El/optimization/bfgs.hpp

```diff
@@ -47,18 +47,18 @@ template< typename T, typename Function, typename Gradient>
 T lineSearch( const Function& f, const Gradient& gradient,
               const DistMatrix<T>& g, Int D,
               const DistMatrix<T>& x0, const DistMatrix<T>& p,
-              Int maxIter=100, T c1=1e-4, T c2=0.9){
+              Int maxIter=100, T c1=Pow(limits::Epsilon< Base<T> >(), Base<T>(1)/Base<T>(4)), T c2=T(9)/T(10)){
 
         if( c1 > c2) { std::swap(c1, c2); }
 
         T f0 = f(x0);
-        DistMatrix<T> g2(D, 1);
+        DistMatrix<T> g2(D, 1, x0.Grid());
         T f0_dash = Dot(p,g);
-        T  alpha = 1;
-        T  alpha_prev = 0;
-        T  alphaMax = 1e3;
-        T fvalPrev = 0;
-        T fval = 0;
+        T  alpha(1);
+        T  alpha_prev(0);
+        T  alphaMax(1e3);
```

Comment:
It is moreso to call attention to the fact that there is a conversion happening. This is fine in this case since double-precision represents 1e3 exactly.